gomoku.py

```diff
@@ -1,46 +1,134 @@
+from abc import ABCMeta, abstractmethod
 import numpy as np
 
 class GomokuBoard:
-    NUM_REQUIRED = 5
-    DIRECTIONS = np.array([[1,0], [1,1], [0,1], [-1,1], [-1,0], [-1,-1], [0,-1], [1,-1]])
+    __NUM_REQUIRED = 5
+    __DIRECTIONS = np.array([[1,0], [1,1], [0,1], [-1,1], [-1,0], [-1,-1], [0,-1], [1,-1]])
     
     def __init__(self, size=19):
-        self.size = size
-        self.board = np.zeros((self.size, self.size), np.int8)
-    
-    def is_on_board(self, coord):
-        return coord[0] >= 0 and coord[0] < self.size \
-            and coord[1] >= 0 and coord[1] < self.size
-    
-    def get_piece(self, coord):
-        if not self.is_on_board(coord):
-            print(f'Invalid coordinate: {coord}')
-            return 0
-        return self.board[coord[0], coord[1]]
-    
+        self.__size = size
+        self.__board = np.zeros((self.__size, self.__size), np.int8)
+
+    def get_board(self):
+        return self.__board
+
+    def get_size(self):
+        return self.__size
+
     def add_piece(self, coord, side):
+        """
+        Place a piece on the board
+
+        Parameters
+        ----------
+        coord : np.array
+            coordinate to place the piece
+        side : int
+            side of the piece
+
+        Returns
+        -------
+        success: bool
+            validity of the move
+        winner: int
+            winning side if it exists, -1 if tied, 0 otherwise
+        board: np.array
+            state of the board after the move
+        """
+
+        if not self.__is_on_board(coord) or self.__get_piece(coord) > 0:
+            print(f'Invalid coordinate: {coord}')
+            return False, 0, self.__board
+
         print(f'Placing {side} at {coord}')
-        self.board[coord[0], coord[1]] = side
+        self.__board[coord[0], coord[1]] = side
```

Comment:
Shouldn't thus be `self.__board[coord[1], coord[0]] = side `?
The x and y-axis are flipped now.

Reply:
You're right, fixed it!